doc/internal/man3/OPTIONS.pod

```diff
@@ -0,0 +1,301 @@
+
+    OPT_PARAMETERS()
+    {OPT_PARAM_STR, 1, '-', "Parameters:\n"}
+
+Every "option" after after this should contain the parameter and 
+the help string:
+
+    {"text", 0, 0, "Words to display (optional)"},
+
+=head2 Functions
+
+The opt_init() function takes the "argc, argv" arguments given to main() and
+a pointer to the list of options. It returns the simple program
+name, as defined by opt_progname().
+
+The opt_progname() function takes the full pathname, C<argv[0]>, and returns
+the simple short name of the executable, to be used for error messages and
+the like.  The opt_appname() functions takes the "application" name (such
+as the specific command from L<openssl(1)> and appends it to the program
+name. This function should only be called once.  Once set, opt_getprog()
+also returns the value.
+
+Once opt_init() has been called, opt_next() can be called in a loop to
+fetch each option in turn. It returns B<-1>, or B<OPT_EOF> when the
+end of arguments has been reached. This is typically done like this:
+
+    prog = opt_init(argc, argv, my_options);
+    while ((o = opt_next()) != OPT_EOF) {
+        switch (o) {
+        case OPT_EOF:
+        case OPT_ERR:
+    opthelp:
+            fprintf(stderr, "%s: Use -help for summary\n", prog);
+            exit(1);
+        case OPT_HELP:
+            opt_help(my_options);
+            exit(0);
+        ...other options...
+        }
+    }
+
+The opt_help() function takes a list of option definitions and prints a
+nicely-formatted output.
+
+Within the option parsing loop, opt_flag() returns the option,
+without any leading hyphens. The opt_arg() function returns
+the option's value, if there is one.
+
+In an option list, there can be at most one option with the empty string.
+This is a "wildcard" or "unknown" option. For example, it allows an
+option to be be taken as digest algorithm, like B<-sha1>. The
+function opt_cipher() takes the specified B<name> and fills in
+the cipher into B<cipherp>.  The function opt_md() does the same
+thing for message digest.
+
+There are a several useful functions for parsing numbers.  These are
+opt_int(), opt_long(), opt_ulong(), opt_imax(), and opt_umax().  They all
+take C<0x> to mean hexadecimal and C<0> to mean octal, and will do the
+necessary range-checking. They return B<1> if successful and fill in the
+C<result> pointer with the value, or B<0> on error. Note that opt_next()
+will also do range-check on the argument if the appropriate B<valtype>
+field is specified for the option. This means that error-checking inside
+the C<switch> C<case> can often be elided.
+
+The opt_isdir() function returns B<1> if the specified B<name> is
+a directory, or B<0> if not. The opt_format() function takes a string value,
+such as used in with the B<-informat> or related flags, and fills
+the value from the constants in F<fmt.h> file.
+
+The opt_string() function checks that B<name> appears in the
+B<NULL>-terminated array of strings. It returns B<1> if found,
```

Comment:
Remove `B<>` around NULL. Also I am not sure about bolding the constants such as 0 and 1.

Reply:
Well it's not obvious :)
```
; g grep NULL doc | grep 'B<'|wc
    461    4893   43517
; g grep NULL doc | grep -v 'B<'|wc
   1362   11938  117044
```

But I removed the `B<>` around NULL and the 0/1 return values.